### PR TITLE
Space laser Manny

### PR DESCRIPTION
**Pull request by MannyB6**
The weapon is able to be switched to the space laser now by just calling the laser ready function which will return true every time as soon as a ship is sunk.

### CODE REVIEW

File: src/main/java/cs361/battleships/models/Board.java [hunk 1]
```diff
@@ -52,12 +59,37 @@ public boolean placeShip(Ship ship, int x, char y, boolean isVertical) {
...
+//		Result attackResult2 = attack(new Square(x, y));
+//		attacks2.add(attackResult2);
+
 		return attackResult;
 	}
 
+
+	//used for the second weapon to check for submerged
+//	private Result attack2(int x, char y){
+//		Result attackResult2 = attack(new Square(x, y));
```
Comment (PembleC) [inline src/main/java/cs361/battleships/models/Board.java]: why all just comments? does it not work?

File: src/test/java/cs361/battleships/models/BoardTest.java [hunk 2]
```diff
@@ -101,4 +101,31 @@ public void isCQHit2() {
         board.attack(3, 'C');
         assertFalse(square1.isHitCQ());
     }
+
+    @Test
```
Comment (PembleC) [inline src/test/java/cs361/battleships/models/BoardTest.java]: Tests are good!  Are we still above 80%

File: src/test/java/cs361/battleships/models/ShipTest.java [hunk 3]
```diff
@@ -241,6 +241,59 @@ public void testCQHitISSetSUBMARINE() {
...
+        minesweeper.place('A', 1, false);
+        assertFalse(square.isHitCQ());
+
+        var result = minesweeper.attack(1,'A');
+        var test = minesweeper.getShipSunk();
+        //assertEquals(AtackStatus.SUNK, result.getResult());
+        assertEquals(true, test);
+    }
+
+//    @Test
```
Comment (PembleC) [inline src/test/java/cs361/battleships/models/ShipTest.java]: commenting out tests isn't the best way to solve them ;)

File: src/main/java/cs361/battleships/models/Board.java (hunk 1 above)
Comment (MannyB6) [reply src/main/java/cs361/battleships/models/Board.java]: Forgot to delete them, I took a different approach.

File: src/test/java/cs361/battleships/models/BoardTest.java (hunk 2 above)
Comment (MannyB6) [reply src/test/java/cs361/battleships/models/BoardTest.java]: Test looks like they haven't changed from master.

File: src/test/java/cs361/battleships/models/ShipTest.java (hunk 3 above)
Comment (MannyB6) [reply src/test/java/cs361/battleships/models/ShipTest.java]: Haha I moved the test in ship to board, forgot to delete it!

**Comment by PembleC** [comment]
Clean code is good code. My only suggestion would be to clean up the comments and commit again. Once that's done and others have commented, lets merge.

**Comment by MannyB6** [comment]
> Clean code is good code. My only suggestion would be to clean up the comments and commit again. Once that's done and others have commented, lets merge.

Cleaned up the code.

**Comment by PembleC** [comment]
Looks Great

File: src/test/java/cs361/battleships/models/BoardTest.java [hunk 4]
```diff
@@ -101,4 +101,29 @@ public void isCQHit2() {
...
```
Comment (reinscha) [inline src/test/java/cs361/battleships/models/BoardTest.java]: Good looking tests!

File: src/main/java/cs361/battleships/models/Ship.java [hunk 5]
```diff
@@ -210,6 +213,9 @@ public Result attack(int x, char y) {
 		else if(attackedSquare.isCQ() && attackedSquare.isHitCQ()){
 			var result = new Result(attackedLocation);
 			attackedSquare.hit();
+
+			setShipSunk(true);//added so a ship has been sunk and the counter is now true
```
Comment (reinscha) [inline src/main/java/cs361/battleships/models/Ship.java]: Good addition here.

**Pull request merged**